GeoLlama/config.py

```diff
@@ -0,0 +1,177 @@
+## Date last modified : 03-May-2024      ##
+###########################################
+
+import os
+
+import typing
+from pathlib import Path
+import multiprocessing as mp
+
+from GeoLlama import objects
+
+
+def generate_config(output_path):
+    """
+    Generates default configuration file
+
+    Args:
+    output_path (str) : Path to output yaml file
+    """
+    import ruamel.yaml
+
+    config_str = """ \
+# Essential parameters
+data_path: null
+pixel_size_nm: null
+
+# Optional parameters
+binning: 0
+autocontrast: True
+adaptive: True
+bandpass: False
+num_cores: 1
+output_csv_path: null
+output_star_path: null
+output_mask: True
+
+# Analytics parameters
+thickness_lower_limit: 120
+thickness_upper_limit: 300
+thickness_std_limit: 15
+xtilt_std_limit: 5
+displacement_limit : 25
+displacement_std_limit : 5
+"""
+
+    yaml = ruamel.yaml.YAML()
+    data = yaml.load(config_str)
+
+    yaml.default_flow_style=False
+    yaml.dump(data, Path(output_path))
+
+
+def read_config(
+        config_fname: str
+) -> objects.Config:
+    """
+    Parse GeoLlama config YAML file
+
+    Args:
+    config_fname (str) : Path to config file
+
+    Returns:
+    Config
+    """
+    import ruamel.yaml
+
+    if not os.path.exists(config_fname):
+        raise IOError(f"{config_fname} does not exist.")
+
+    yaml = ruamel.yaml.YAML()
+    config_dict = yaml.load(Path(config_fname))
+
+    # Initialise a Config object
+    params = objects.Config()
+
+    # Check config file has correct dictionary keys
+    for key in params.__dict__.keys():
+        try:
+            params.__setattr__(key, config_dict[key])
+        except ValueError:
+            print(f"{key} keyword missing in config file.")
+
+    return params
+
+
+def objectify_user_input(
+        autocontrast: typing.Optional[bool],
+        adaptive: typing.Optional[bool],
+        bandpass: typing.Optional[bool],
+        data_path: typing.Optional[str],
+        pixel_size_nm: typing.Optional[float],
+        binning: typing.Optional[int],
+        num_cores: typing.Optional[int],
+        output_csv_path: typing.Optional[str],
+        output_star_path: typing.Optional[str],
+        output_mask: typing.Optional[bool],
+
+        thickness_lower_limit: typing.Optional[float],
+        thickness_upper_limit: typing.Optional[float],
+        thickness_std_limit: typing.Optional[float],
+        xtilt_std_limit: typing.Optional[float],
+        displacement_limit : typing.Optional[float],
+        displacement_std_limit : typing.Optional[float],
+) -> objects.Config:
+    """
+    Objectifying user provided input as a Config object
+
+    Args:
+    autocontrast (bool) : Apply autocontrast to slices prior to evaluation
+    adaptive (bool) : Use adaptive sampling for slice evaluation
+    bandpass (bool) : Apply bandpass filter to tomograms prior to evaluation
+    user_path (str) : Path to folder holding all tomograms in batch mode
+    pixel_size (float) : Tomogram pixel size in nm
```

Comment:
Might want to be quite explicit here - if the mdocs say my pixel size is 1 A, and I've reconstructed my tomograms with bin 8, is the `pixel_size` for Geollama supposed to be 0.1 nm or 0.8 nm?

Reply:
It is the input pixel size, so 0.1 in your example. Will change in the code.